File: src/test/scala/repository/eventsourcing/example/port/adapter/persistence/EventSourcedHouseRepository.scala
package repository.eventsourcing.example.port.adapter.persistence

import eventstore.api.EventStore
import repository.eventsourcing.EventSourcedRepository
import repository.eventsourcing.example.domain.{House, HouseRepository}

class EventSourcedHouseRepository(override val eventStore: EventStore) extends EventSourcedRepository[House, String](eventStore) with HouseRepository  {

}

File: ddd-eventsourcing/src/test/java/ddd/repository/eventsourcing/example/port/adapter/persistence/EventSourcedHouseRepository.java
package ddd.repository.eventsourcing.example.port.adapter.persistence;

import ddd.eventstore.EventStore;
import ddd.repository.eventsourcing.EventSourcedRepository;
import ddd.repository.eventsourcing.example.domain.HouseRepository;
import ddd.repository.eventsourcing.example.domain.House;

public class EventSourcedHouseRepository extends EventSourcedRepository<House, String> implements HouseRepository {
    public EventSourcedHouseRepository(EventStore eventstore) {
        super(eventstore);
    }
}
